core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigration.scala

```diff
@@ -289,22 +288,20 @@ trait MultiregionalBucketMigrationHistory extends DriverComponent with RawSqlQue
         .as[Int]
         .map(_.head > 0)
 
-    final def scheduleAndGetMetadata: WorkspaceName => ReadWriteAction[MultiregionalBucketMigrationMetadata] =
-      (schedule _) >=> getMetadata
+    final def scheduleAndGetMetadata
+      : (Workspace, Option[String]) => ReadWriteAction[MultiregionalBucketMigrationMetadata] =
+      schedule(_, _).flatMap(getMetadata)
 
-    final def schedule(workspaceName: WorkspaceName): ReadWriteAction[Long] =
+    final def schedule(workspace: Workspace, location: Option[String]): ReadWriteAction[Long] =
       for {
-        workspaceOpt <- workspaceQuery.findByName(workspaceName)
```

Comment:
We always need to fetch the workspace before trying to schedule it for migration because we need the workspace's bucket name, so `schedule` now takes a `Workspace` and it's up to the `BucketMigrationService` to fetch the workspace first. Has the added benefit of not loading each workspace twice when scheduling all workspaces in a billing project.